core/domain/exp_domain.py

```diff
@@ -2499,3 +2499,541 @@ def add_contribution_by_user(self, contributor_id):
+        STATE_PROPERTY_INTERACTION_ANSWER_GROUPS]
+
+    # PROPERTIES_RELATED_TO_ANSWER_GROUPS: List of the properties
+    # related (affected by or affecting) answer groups. This list
+    # can be changed when any new property is added or deleted which
+    # affects or is affected by answer groups.
+    PROPERTIES_RELATED_TO_ANSWER_GROUPS = [
+        STATE_PROPERTY_INTERACTION_SOLUTION,
+        STATE_PROPERTY_RECORDED_VOICEOVERS,
+        STATE_PROPERTY_INTERACTION_CUST_ARGS]
+
+    # PROPERTIES_RELATED_TO_SOLUTION: List of the properties
+    # related (affected by or affecting) solution. This list
+    # can be changed when any new property is added or deleted which
+    # affects or is affected by solution.
+    PROPERTIES_RELATED_TO_SOLUTION = [
+        STATE_PROPERTY_INTERACTION_ANSWER_GROUPS,
+        STATE_PROPERTY_RECORDED_VOICEOVERS,
+        STATE_PROPERTY_INTERACTION_CUST_ARGS]
+
+    # PROPERTIES_RELATED_TO_VOICEOVERS: List of the properties
+    # related (affected by or affecting) voiceovers. This list
+    # can be changed when any new property is added or deleted which
+    # affects or is affected by voiceovers.
+    PROPERTIES_RELATED_TO_VOICEOVERS = [
+        STATE_PROPERTY_CONTENT,
+        STATE_PROPERTY_INTERACTION_SOLUTION,
+        STATE_PROPERTY_INTERACTION_HINTS,
+        STATE_PROPERTY_WRITTEN_TRANSLATIONS,
+        STATE_PROPERTY_INTERACTION_ANSWER_GROUPS,
+        STATE_PROPERTY_INTERACTION_DEFAULT_OUTCOME,
+        STATE_PROPERTY_INTERACTION_CUST_ARGS]
+
+    def is_change_list_mergeable(
+            self, change_list,
+            exp_at_change_list_version, current_exploration):
+        """Checks whether the change list from the old version of an
+        exploration can be merged on the latest version of an exploration.
+
+        Args:
+            change_list: list(ExplorationChange). List of the changes made
+                by the user on the frontend, which needs to be checked
+                for mergeability.
+            exp_at_change_list_version: obj. Old version of an exploration.
+            current_exploration: obj. Exploration on which the change list
+                is to be applied.
+
+        Returns:
+            dict. A response dict for the change_list.
+            The response dict has two keys:
+                - 'changes_are_mergeable': boolean. Whether the given change
+                    list is mergeable on the current_exploration or not.
+                - 'send_email': boolean. Whether we need to send the change
+                    list to the admin to review for the future improvement
+                    of the cases to merge the change list.
+        """
+
+        added_state_names = self.added_state_names
+        deleted_state_names = self.deleted_state_names
+        new_to_old_state_names = self.new_to_old_state_names
+        changed_properties = self.changed_properties
+        changed_translations = self.changed_translations
```

Comment:
Do not have separate variables for the object properties*

Reply:
Removed it.